File: bench/cases/csv-rebuild-100k/main.go
package main

import (
	"fmt"
	"strings"
)

func rebuild(line string) int64 {
	var total int64 = 0
	for _, part := range strings.Split(line, ",") {
		s := part + "|"
		total += int64(s[0])
	}
	return total
}

func main() {
	var total int64 = 0
	var n int64 = 100_000
	for i := int64(0); i < n; i++ {
		total += rebuild("alpha,beta,gamma,delta,epsilon,zeta")
	}
	fmt.Println(total)
}

File: bench/cases/csv-rebuild-100k/main.rs
fn rebuild(line: &str) -> i64 {
    let mut total: i64 = 0;
    for part in line.split(',') {
        let s = format!("{}|", part);
        total += s.as_bytes()[0] as i64;
    }
    total
}

fn main() {
    let mut total: i64 = 0;
    let n: i64 = 100_000;
    for _ in 0..n {
        total += rebuild("alpha,beta,gamma,delta,epsilon,zeta");
    }
    println!("{}", total);
}
